### PR TITLE
fix(clients/go): print activate jobs ouput as valid json

### LINKED ISSUE #5888: zbctl activate jobs ouput json is not correct
**Describe the bug**
`zbctl` display activate jobs as a bad formatted json when there is more than one job. So its very hard to use it programmatically.

**To Reproduce**
Run `zbctl activate jobs` command and  manage to start more than one job in a raw...

If possible add a minimal reproducer code sample
 - when using the Go client: https://github.com/zeebe-io/zeebe-docker-compose/blob/master/bin/zbctl
- output is like this:

```json
{
  "key": 2251799813699496,
  "type": "madam_scanfolder",
  "workflowInstanceKey": 2251799813699491,
  "bpmnProcessId": "test_7KB9e9SbraYtQ7vPzy0jKhXoZxmq8rNH",
  "workflowDefinitionVersion": 1,
  "workflowKey": 2251799813699488,
  "elementId": "scanfolder_ba",
  "elementInstanceKey": 2251799813699495,
  "customHeaders": "{\"source\":\"/mnt/diskstation2/MEDIAS/data/madam/tests/test_scanfolder\",\"recursive\":\"False\"}",
  "worker": "zbctl",
  "retries": 3,
  "deadline": 1605931882895,
  "variables": "{}"
}
{
  "key": 2251799813699507,
  "type": "madam_scanfolder",
  "workflowInstanceKey": 2251799813699501,
  "bpmnProcessId": "test_u32QH7qQVeMRxLFKk1jOn6NMWd906NS0",
  "workflowDefinitionVersion": 1,
  "workflowKey": 2251799813699499,
  "elementId": "scanfolder_ba",
  "elementInstanceKey": 2251799813699506,
  "customHeaders": "{\"source\":\"/mnt/diskstation2/MEDIAS/data/madam/tests/test_scanfolder\",\"recursive\":\"False\"}",
  "worker": "zbctl",
  "retries": 3,
  "deadline": 1605931882895,
  "variables": "{}"
}
```

**Expected behavior**
- `zbctl` activate jobs output should be parsed correctly by any json parser.
- We should have **brackets** to define a list, and jobs should be separated with **commas**, like this:

```json
[{
  "key": 2251799813699496,
  "type": "madam_scanfolder",
  "workflowInstanceKey": 2251799813699491,
  "bpmnProcessId": "test_7KB9e9SbraYtQ7vPzy0jKhXoZxmq8rNH",
  "workflowDefinitionVersion": 1,
  "workflowKey": 2251799813699488,
  "elementId": "scanfolder_ba",
  "elementInstanceKey": 2251799813699495,
  "customHeaders": "{\"source\":\"/mnt/diskstation2/MEDIAS/data/madam/tests/test_scanfolder\",\"recursive\":\"False\"}",
  "worker": "zbctl",
  "retries": 3,
  "deadline": 1605931882895,
  "variables": "{}"
},
{
  "key": 2251799813699507,
  "type": "madam_scanfolder",
  "workflowInstanceKey": 2251799813699501,
  "bpmnProcessId": "test_u32QH7qQVeMRxLFKk1jOn6NMWd906NS0",
  "workflowDefinitionVersion": 1,
  "workflowKey": 2251799813699499,
  "elementId": "scanfolder_ba",
  "elementInstanceKey": 2251799813699506,
  "customHeaders": "{\"source\":\"/mnt/diskstation2/MEDIAS/data/madam/tests/test_scanfolder\",\"recursive\":\"False\"}",
  "worker": "zbctl",
  "retries": 3,
  "deadline": 1605931882895,
  "variables": "{}"
}]
```
- To have a consistent ouput, even when there is only one job, it should be in a json list.

**Environment:**
 - OS: Linux
 - Zeebe Version: 0.23.7
 - Configuration: [e.g. exporters etc.]

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

In previous version the output of zbctl activate job was not really json compilant. Furthermore there was additional output to make it human readable, which is not necessary and not consistent with the other commands we have. The additional output makes it also harder to process it further with other commands

**Previous output:**

```sh
[zell ~/ ns:testbench]$ zbctl --insecure activate jobs --maxJobsToActivate 2 benchmark-task
2020/12/04 21:15:23 Activated 2 for type benchmark-task
2020/12/04 21:15:23 Job 1 / 2
{
  "key": 2251799813685257,
  "type": "benchmark-task",
  "workflowInstanceKey": 2251799813685252,
  "bpmnProcessId": "benchmark",
  "workflowDefinitionVersion": 1,
  "workflowKey": 2251799813685250,
  "elementId": "task",
  "elementInstanceKey": 2251799813685256,
  "customHeaders": "{}",
  "worker": "zbctl",
  "retries": 3,
  "deadline": 1607113223419,
  "variables": "{}"
}
2020/12/04 21:15:23 Job 2 / 2
{
  "key": 2251799813685263,
  "type": "benchmark-task",
  "workflowInstanceKey": 2251799813685258,
  "bpmnProcessId": "benchmark",
  "workflowDefinitionVersion": 1,
  "workflowKey": 2251799813685250,
  "elementId": "task",
  "elementInstanceKey": 2251799813685262,
  "customHeaders": "{}",
  "worker": "zbctl",
  "retries": 3,
  "deadline": 1607113223419,
  "variables": "{}"
}
```

**New version:**

```sh
[zell dist/ ns:testbench]$ ./zbctl --insecure activate jobs --maxJobsToActivate 2 benchmark-task
[
  {
    "key": 2251799813685270,
    "type": "benchmark-task",
    "workflowInstanceKey": 2251799813685265,
    "bpmnProcessId": "benchmark",
    "workflowDefinitionVersion": 1,
    "workflowKey": 2251799813685250,
    "elementId": "task",
    "elementInstanceKey": 2251799813685269,
    "customHeaders": "{}",
    "worker": "zbctl",
    "retries": 3,
    "deadline": 1607113327063,
    "variables": "{}"
  },
  {
    "key": 2251799813685276,
    "type": "benchmark-task",
    "workflowInstanceKey": 2251799813685271,
    "bpmnProcessId": "benchmark",
    "workflowDefinitionVersion": 1,
    "workflowKey": 2251799813685250,
    "elementId": "task",
    "elementInstanceKey": 2251799813685275,
    "customHeaders": "{}",
    "worker": "zbctl",
    "retries": 3,
    "deadline": 1607113327063,
    "variables": "{}"
  }
]
```

**What I did:**

 * the result of the command is directly serialized as json
 * if the jobs result is empty then an empty json array is printed - the json marshall would return an empty string, which we dont want
 * the additional log outputs are removed
 * two new tests are added for empty response and multiple job activation

## Related issues

closes https://github.com/zeebe-io/zeebe/issues/5888

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: clients/go/cmd/zbctl/main_test.go [hunk 1]
```diff
@@ -95,10 +95,22 @@ var tests = []struct {
...
+		name:       "single activate job",
 		setupCmds:  []string{"--insecure deploy testdata/job_model.bpmn", "--insecure create instance jobProcess"},
-		cmd:        "--insecure activate jobs jobType",
-		goldenFile: "testdata/activate_job.golden",
+		cmd:        "--insecure activate jobs jobType --maxJobsToActivate 1",
+		goldenFile: "testdata/single_activate_job.golden",
+	},
+	{
+		name: "double activate job",
+		// we deploy on the end again to spent more time in setup phase to avoid a race condition, that we can activate more jobs then one
```
Comment (npepinpe) [inline clients/go/cmd/zbctl/main_test.go:110]: This isn't obvious to me, can you elaborate? By deploying this means we spend more time in setup, so we don't have to wait for the jobs to be created?

File: clients/go/cmd/zbctl/main_test.go [hunk 2]
```diff
@@ -118,16 +130,16 @@ func TestZbctlWithInsecureGateway(t *testing.T) {
 		})
 }
 
-func (s *integrationTestSuite) TestCommonCommands() {
+func (suite *integrationTestSuite) TestCommonCommands() {
```
Comment (npepinpe) [inline clients/go/cmd/zbctl/main_test.go:133]: hehe, welcome to the world of go, where all variables are a single letter :joy:

File: clients/go/cmd/zbctl/main_test.go (hunk 1 above)
Comment (npepinpe) [reply clients/go/cmd/zbctl/main_test.go:110]: If that's correct, then wouldn't setting a longer time out on activate jobs work? Or is that not possible via zbctl?

File: clients/go/cmd/zbctl/main_test.go (hunk 2 above)
Comment (Zelldon) [reply clients/go/cmd/zbctl/main_test.go:133]: Oh is this convention?🙈😅 I hope not 😅

Comment (npepinpe) [reply clients/go/cmd/zbctl/main_test.go:133]: https://github.com/golang/go/wiki/CodeReviewComments#variable-names :man_shrugging: I think I made it sound worse than it is, I kind of get the reasoning, but I guess it mostly works if you're disciplined and keep your functions pretty short.

**Review by npepinpe — CHANGES_REQUESTED** [review]
Can we update this based on https://github.com/zeebe-io/zeebe/pull/5943 and https://github.com/zeebe-io/zeebe/issues/5975?

So we leave the output as is by default (`-o human`), but we can add `-o json` and output using JSONPB.

Sorry this comes after the PR was opened

**Comment by Zelldon** [comment]
Hey  @npepinpe I think I disagree. First I don't want to get this PR bigger then necessary, I would like to keep the scope small. For me it is a bug fix and to make the output consistent to all other commands, which is btw json. This brings me to the second point. I see not really a value in having this output flag on the commands, since all print json already. Json is also human readable so what would be the output you expect for `-o human` and then the question is this really necessary/needed?

**Comment by npepinpe** [comment]
Human in this case refers to the fact it's not meant to be consumed for scripts/by machines, not that the other formats cannot be read by humans. From my point of view, having a flag to specify the output is quite useful in the long term, but that's probably a discussion we should have in the other issue.

Regarding this PR, this means adding a breaking a change - I disagree this is a bug, since it was clearly meant to perform in this very specific way. It's inconsistent, but I wouldn't call it a bug. However it's probably not a big breaking change - you only lose a bit of extra output, but no real information is lost. Might break somebody's scripts, but the new way is easier for them.

So, tl;dr, you were correct to push back, let's go with these changes for now, and the other issues can be implemented afterwards.

**Comment by Zelldon** [comment]
Just for the record I'm not against the output flag I think it makes sense to have different options like yaml, xml etc. I just question the human output for the command responses, but as you said we can discuss this in a different place/time.

So what is left, needs to be done?

* bringing back the `s` naming 😅🙈
 * trying out whether it works with an higher timeout, which is an good idea i think

**Comment by npepinpe** [comment]
Nothing, even the s stuff is fine imo - it's a convention, not a rule :sweat_smile: 

Yeah let's try the time out stuff, I think that makes more sense than an additional deployment - what if Zeebe becomes so fast it doesn't slow down enough :wink: ?

File: clients/go/cmd/zbctl/main_test.go (hunk 1 above)
Comment (Zelldon) [reply clients/go/cmd/zbctl/main_test.go:110]: Ok I tried with requestTimeout 30s, but doesn't work. The thing is that only one job exist when the activate command is send (without the second deployment). This job is immediately returned. In general this is fine because this is just the max value we are setting, but in this test we want explicitly 2 because we want to test the output. This is the reason why I added the second deployment to have more time such that the second job is available. Sleep would also be ok but I dont know where to put this, since the tests are highly parameterized.

**Comment by MiguelPires** [comment]
If we want to keep the PR scope limited then I'd say that's an additional reason for not going out of our way to break conventions such as keeping the receiver names short.

**Comment by Zelldon** [comment]
If it is a convention then we should follow it, even if I don't like it tbh. I have renamed the variable back to `s`.

File: clients/go/cmd/zbctl/main_test.go (hunk 1 above)
Comment (npepinpe) [reply clients/go/cmd/zbctl/main_test.go:117]: I see what you mean - there's no way for us to wait until both jobs are created before actually activating them. My main worry is this has potential to be flaky, but I wouldn't block this for now.

**Review by npepinpe — APPROVED** [review]

**Comment by npepinpe** [comment]
bors r+

I force pushed to rebase-squash, hope that's alright with you.
